web-client/src/http/resources/auth.ts

```diff
@@ -11,15 +11,28 @@ export const login = (request: IHTTPRequest) => {
   return request.execute(config).then((response: any) => response.data);
 };
 
-export const facebookLoginWithFirebase = async (): Promise<firebase.auth.UserCredential> => {
+export const facebookLoginWithFirebasePopUp = async (): Promise<
+  string | undefined
+> => {
   const provider = new firebase.auth.FacebookAuthProvider();
   const result = await firebaseAuth.signInWithPopup(provider);
-  return result;
+  const token = await result.user?.getIdToken();
```

Comment:
In these request we could just continue using the promise, since `injectRequestMiddleware` already handles the `await` for us. I think this way is a bit cleaner. What do you think?

Reply:
Basically, just like we would do with axios.

Reply:
oh yes, that makes sense.. now I understand why the error actions weren't being dispatched. 
However, only the promise for the token should be handled by it. I will make the change in PR.
I did go through the middleware, but maybe because I was trying to solve another issue. this particular point of view skipped my mind.
I will make the change

Reply:
I have made this change now